lte/gateway/c/oai/tasks/ngap/ngap_state_manager.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/****************************************************************************
+  Source      ngap_state_manager.cpp
+  Date        2020/07/28
+  Author      Ashish Prajapati
+  Subsystem   Access and Mobility Management Function
+  Description Defines NG Application Protocol Messages
+
+*****************************************************************************/
+
+#include "ngap_state_manager.h"
+#include "bstrlib.h"
+#include "log.h"
+typedef unsigned int uint32_t;
+namespace {
+constexpr char NGAP_GNB_COLL[]             = "ngap_gNB_coll";
+constexpr char NGAP_AMF_ID2ASSOC_ID_COLL[] = "ngap_amf_id2assoc_id_coll";
+constexpr char NGAP_IMSI_MAP_TABLE_NAME[]  = "ngap_imsi_map";
+}  // namespace
+
+using magma::lte::oai::Ngap_UeDescription;
+
+namespace magma5g {
+
+NgapStateManager::NgapStateManager() : max_gnbs_(0), max_ues_(0) {}
+
+NgapStateManager::~NgapStateManager() {
+  free_state();
+}
+
+NgapStateManager& NgapStateManager::getInstance() {
+  static NgapStateManager instance;
+  return instance;
+}
+
+void NgapStateManager::init(
+    uint32_t max_ues, uint32_t max_gnbs, bool use_stateless) {
+  log_task              = LOG_NGAP;
+  table_key             = NGAP_STATE_TABLE;
+  task_name             = NGAP_TASK_NAME;
+  persist_state_enabled = use_stateless;
```

Comment:
Set this to false as we are not persisting state to Redis for 5G core at the moment
```suggestion
  persist_state_enabled = false;
```